src-opam/opam.ml

```diff
@@ -237,12 +236,33 @@ let create_opam_branches opam_hashes =
         public_name = "opam-2.1";
         aliases = [];
       };
+      {
+        branch = "2.2";
+        hash = opam_2_2_hash;
+        enable_0install_solver = true;
+        with_vendored_deps = false;
+        public_name = "opam-2.2";
+        aliases = [];
+      };
       opam_master_branch opam_master_hash;
     ] )
 
 let create_opam_branches_windows opam_hashes =
-  let { opam_master_hash; _ } = opam_hashes in
-  (opam_master_hash, [ opam_master_branch opam_master_hash ])
+  let { opam_2_0_hash; opam_2_1_hash; opam_2_2_hash; opam_master_hash } =
+    opam_hashes
+  in
+  ( opam_master_hash,
+    [
+      {
+        branch = "2.2";
+        hash = opam_2_2_hash;
+        enable_0install_solver = true;
+        with_vendored_deps = true;
+        public_name = "opam-2.2";
+        aliases = [];
+      };
```

Comment:
Maybe this could be factored with the other definition above?

Reply:
I've updated the code, so it is defined once and used a filter to remove the branches we don't want to build under Windows.